### PR TITLE
Feature/import documents batch threads

### PR DESCRIPTION
**Pull request by rkhaja**
This pull request addresses enhancements requested in #273 
Please review.

### CODE REVIEW

**Comment by hkernbach** [comment]
Hey @rkhaja, first of all, thank you very much for contributing. The code itself looks good to me. We need a JUnit test to test that feature, then we're able to merge. 

If you could also add a test, it would be great. Otherwise, we will add one as soon as we can.

**Comment by rkhaja** [comment]
@hkernbach Thanks!  I will of course write the JUnit test(s) (hopefully very soon) ... I wanted to be sure that the this code met your coding standards before I spent effort to write the tests.

**Comment by rkhaja** [comment]
I've added the following test:
```
	@Test
	public void importDocumentsBatchSizeNumThreads() {
		final Collection<BaseDocument> values = new ArrayList<BaseDocument>();
		for (int i = 1; i <= 100; i++) {
			values.add(new BaseDocument(String.valueOf(i)));
		}
		int batchSize = 5;
		int numThreads = 4;
		final Collection<DocumentImportEntity> docsList = db.collection(COLLECTION_NAME).importDocuments(values,
				new DocumentImportOptions(), batchSize, numThreads);
        assertThat(docsList.size(), is(values.size() / batchSize));
		for (final DocumentImportEntity docs : docsList) {
			assertThat(docs, is(notNullValue()));
			assertThat(docs.getCreated(), is(batchSize));
			assertThat(docs.getEmpty(), is(0));
			assertThat(docs.getErrors(), is(0));
			assertThat(docs.getIgnored(), is(0));
			assertThat(docs.getUpdated(), is(0));
			assertThat(docs.getDetails(), is(empty()));
		}
	}
```
Essentially it is a copy of the basic `importDocuments()` test except that iterates through individual DocumentImportEntity objects.
```
	@Test
	public void importDocuments() {
		final Collection<BaseDocument> values = new ArrayList<BaseDocument>();
		values.add(new BaseDocument());
		values.add(new BaseDocument());
		values.add(new BaseDocument());
		final DocumentImportEntity docs = db.collection(COLLECTION_NAME).importDocuments(values);
		assertThat(docs, is(notNullValue()));
		assertThat(docs.getCreated(), is(values.size()));
		assertThat(docs.getEmpty(), is(0));
		assertThat(docs.getErrors(), is(0));
		assertThat(docs.getIgnored(), is(0));
		assertThat(docs.getUpdated(), is(0));
		assertThat(docs.getDetails(), is(empty()));
	}
```
I hope the test is sufficient but if it is not please let me know.

**Comment by hkernbach** [comment]
Thank you for contributing, will merge your PR now. This will be then part of our next ArangoDB Java Driver release (Version: 5.0.8).

**Comment by rkhaja** [comment]
Thank you!

File: src/main/java/com/arangodb/internal/ArangoCollectionImpl.java [hunk 1]
```diff
@@ -105,6 +112,32 @@ public DocumentImportEntity importDocuments(final Collection<?> values, final Do
...
+		List<DocumentImportEntity> documentImportEntityList = new ArrayList<>();
+		for (CompletableFuture<DocumentImportEntity> completableFuture : completableFutureList) {
+			DocumentImportEntity documentImportEntity = null;
+			try {
+				documentImportEntity = completableFuture.get();
+			} catch (InterruptedException | ExecutionException e) {
+				throw new ArangoDBException(e);
+			}
+			documentImportEntityList.add(documentImportEntity);
+		}
```
Comment (rkhaja) [inline src/main/java/com/arangodb/internal/ArangoCollectionImpl.java]: @hkernbach I noticed a small problem.  
RIght before the return statement on this [line](https://github.com/arangodb/arangodb-java-driver/blob/master/src/main/java/com/arangodb/internal/ArangoCollectionImpl.java#L138l)
we need to shutdown the executorService.
```
        executorService.shutdown();
```
If another pull request is needed, please let me know.

**Comment by rkhaja** [comment]
@hkernbach I noticed a small problem.  
RIght before the return statement on this [line](https://github.com/arangodb/arangodb-java-driver/blob/master/src/main/java/com/arangodb/internal/ArangoCollectionImpl.java#L138l)
we need to shutdown the executorService.
```
        executorService.shutdown();
```
If another pull request is needed, please let me know.

File: src/main/java/com/arangodb/internal/ArangoCollectionImpl.java (hunk 1 above)
Comment (hkernbach) [reply src/main/java/com/arangodb/internal/ArangoCollectionImpl.java]: Will fix that, thanks for spotting :)

**Pull request merged**